### ISSUE TITLE
autoproj versions does not record branch properly

### ISSUE THREAD

**Issue by jakobs** [rock-core/autoproj#48]
I have an override in the overrides.d for base/types
```
overriden in main configuration
      branch: update_logging
      commit: 03ed628700601823e42ad3126a614c3b6952902e`
```

my actual HEAD is on master, but when I do 
`autoproj versions base/types`
I get
```
- base/types:
    branch: update_logging
    tag: 
    commit: 03ed628700601823e42ad3126a614c3b6952902e
```

the commitid is correct, but not the branch.

**Comment by doudou** [comment]
This is currently the expected behaviour. autoproj does not know what you are pulling from anyways, so the only sane thing he can do is use the branch recorded in the autoproj configuration.

I retagged to post-versioning because this is part of "making versioning work under any kind of local modifications to the git repository" rather than "support snapshotting for rock releases".

**Comment by jakobs** [comment]
IMO, the question is if it should even record the branch, if a commit id is given. And if, why not take the branch of the current HEAD?

**Comment by doudou** [comment]
This is the expected behaviour *right now*: autoproj only snapshots against the current remote branch. The reason is that fetching all remote commits can possibly be pretty expensive, and that the natural mode of operation in autoproj (so far) was to assume that the configured version control should match the reality.

You created half a dozen issues around this particular behaviour, and we both agreed it should change. What I argue now is that it should change *later* in order to get the next rock release rolling.

**Comment by doudou** [comment]
Functionality is implemented in autoproj 2.0
